up_client_zenoh/upclientzenoh.py

```diff
@@ -0,0 +1,463 @@
+"""
+SPDX-FileCopyrightText: Copyright (c) 2024 Contributors to the Eclipse Foundation
+
+See the NOTICE file(s) distributed with this work for additional
+information regarding copyright ownership.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+SPDX-FileType: SOURCE
+SPDX-License-Identifier: Apache-2.0
+"""
+
+import logging
+from concurrent.futures import Future
+from threading import Lock
+from typing import Dict, Tuple, Any
+
+import zenoh
+from uprotocol.proto.uattributes_pb2 import CallOptions
+from uprotocol.proto.uattributes_pb2 import UAttributes, UMessageType
+from uprotocol.proto.uattributes_pb2 import UPriority
+from uprotocol.proto.umessage_pb2 import UMessage
+from uprotocol.proto.upayload_pb2 import UPayloadFormat, UPayload
+from uprotocol.proto.uri_pb2 import UUri, UAuthority, UEntity
+from uprotocol.proto.ustatus_pb2 import UStatus, UCode
+from uprotocol.rpc.rpcclient import RpcClient
+from uprotocol.transport.builder.uattributesbuilder import UAttributesBuilder
+from uprotocol.transport.ulistener import UListener
+from uprotocol.transport.utransport import UTransport
+from uprotocol.transport.validate.uattributesvalidator import Validators
+from uprotocol.uri.factory.uresource_builder import UResourceBuilder
+from uprotocol.uri.validator.urivalidator import UriValidator
+from zenoh import open as zenoh_open, Subscriber, Session, Encoding, Config, Query, Sample, Value, Queryable
+
+from up_client_zenoh.zenohutils import ZenohUtils
+
+logger = logging.getLogger(__name__)
+
+
+class UPClientZenoh(UTransport, RpcClient):
+
+    def __init__(self, session: Session, subscriber_map: Dict[Tuple[str, UListener], Subscriber],
```

Comment:
Just curious why we need to have these Dicts in the argument.

Reply:
It's not needed as I am passing an empty dictionary from its new method. I've simplified it now and removed all the dicts from the methods, initializing them at the time of creating the UpclientZenoh object.